twisted/python/_release.py

```diff
@@ -58,8 +58,8 @@ def runCommand(args, cwd=None):
     @raise CommandFailed: when the program exited with a non-0 exit code.
     """
     process = Popen(args, stdout=PIPE, stderr=STDOUT, cwd=cwd)
-    stdout = process.stdout.read()
-    exitCode = process.wait()
+    stdout = process.communicate()[0]
+    exitCode = process.poll()
```

Comment:
Popen.communicate (on the line before) until the child process exits.
Popen.poll lets you retrieve the exit status without blocking.

https://docs.python.org/2/library/subprocess.html#subprocess.Popen.communicate

https://docs.python.org/2/library/subprocess.html#subprocess.Popen.poll

On Wednesday, July 20, 2016, Craig Rodrigues <notifications@github.com>
wrote:

> In twisted/python/_release.py
> <https://github.com/twisted/twisted/pull/368#discussion_r71638820>:
>
> > @@ -58,8 +58,8 @@ def runCommand(args, cwd=None):
> >      @raise CommandFailed: when the program exited with a non-0 exit code.
> >      """
> >      process = Popen(args, stdout=PIPE, stderr=STDOUT, cwd=cwd)
> > -    stdout = process.stdout.read()
> > -    exitCode = process.wait()
> > +    stdout = process.communicate()[0]
> > +    exitCode = process.poll()
>
> Why did you change process.wait() to process.poll() ?
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/twisted/twisted/pull/368/files/2c3476b4fba0a52fe44d94929b4e4910182c61ea#r71638820>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAq-97RqRRg25IU-nAs6HmF2TiVoTmwdks5qXtlHgaJpZM4JQ3Pf>
> .
>